make/Images.gmk

```diff
@@ -130,10 +130,16 @@ CDS_DUMP_FLAGS = -Xmx128M -Xms128M
 # Helper function for creating the CDS archives for the JDK and JRE
 #
 # Param1 - VM variant (e.g., server, client, zero, ...)
-# Param2 - _nocoops, or empty
+# Param2 - _nocoops, _coh, _nocoops_coh, or empty
 define CreateCDSArchive
-  $1_$2_DUMP_EXTRA_ARG := $(if $(filter _nocoops, $2),-XX:-UseCompressedOops,)
-  $1_$2_DUMP_TYPE      := $(if $(filter _nocoops, $2),-NOCOOPS,)
+  $1_$2_COOPS_OPTION := $(if $(findstring _nocoops, $2),-XX:-UseCompressedOops)
+  # enable and also explicitly disable coh as needed.
+  ifeq ($(call isTargetCpuBits, 64), true)
```

Comment:
Not all 64-bit platforms currently support COH. Only aarch64 and x86_64 do. Would that be a problem?

Reply:
Okay, fixed and tested. We now only build COH archives when on x64 or on aarch64, and for the latter, only when not crossbuilding.

Tested on Linux x64 (for x64, x32, aarch64 crossbuild) and Mac m1 (aarch64 native)